### PR TITLE
wallet emulator import & API

### PR DESCRIPTION
**Pull request by j-mueller**
* Change UTXO model in `wallet-api` to match the reference implementation in cardano-sl/new-wallet/test/unit where applicable
* Nicer API for `Wallet.Emulator.Types`
* Generate transactions with multiple outputs in emulator tests; add parameters for mockchain size and fee estimation

### CODE REVIEW

File: wallet-api/src/Wallet/Emulator/Types.hs [hunk 1]
```diff
@@ -140,11 +147,32 @@ eval = \case
...
-        case assertion s of
+        case at s of
             Just err -> throwError err
             Nothing  -> pure ()
 
 process :: (MonadState EmulatorState m, MonadError AssertionError m) => Trace a -> m a
 process = interpretWithMonad eval
+
+-- | Interact with a wallet
+walletAction :: Wallet -> EmulatedWalletApi () -> Trace [Tx]
```
Comment (michaelpj) [inline wallet-api/src/Wallet/Emulator/Types.hs]: :clap: I should have done this to begin with.

**Comment by michaelpj** [comment]
(In particular `Data.Set` is exported as abstract, and I'm not sure we'll ever be able to support abstract types)

**Comment by j-mueller** [comment]
> the transaction ins/outs as a set now?

Actually it's only the inputs that are sets, the outputs need to be a list because they are referenced by their index in the list by the consuming transaction

**Comment by michaelpj** [comment]
I think it *probably* doesn't matter if the scripts get a linearized sequence of inputs rather than a set, but worth bearing in mind. Alternatively we could pass them in a simpler non-ordered container type that we define.

**Comment by michaelpj** [comment]
Yeah, we do need to get the same linearisation, because scripts *could* depend on it (although it would probably be a bad idea to!).

**Pull request merged**